Documentation/ApiOverview/RequestLifeCycle/RequestAttributes/FrontendController.rst

```diff
@@ -0,0 +1,25 @@
+..  include:: /Includes.rst.txt
+
+..  index::
+    Request attribute; Frontend controller
+..  _typo3-request-attribute-frontend-controller:
+
+===================
+Frontend controller
+===================
+
+The :php:`frontend.controller` frontend request attribute provides access to the
+:php:`\TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController` object.
+
+Example:
+
+..  code-block:: php
+
+    $frontendController = $request->getAttribute('frontend.controller');
+    $rootline = $frontendController->rootline;
+
+..  important::
+    In former TYPO3 versions you have to retrieve the
```

Comment:
Can we be more specific about the version? Before and now are not clear or time

Reply:
I assume, before v9 as the routing is not there. I did no archeology to check when the attribute 'frontend.controller' started to exist (maybe in v9, maybe starting with v10, but it is available in v11).
If you haven't the request object available in a class you can either get TSFE via $GLOBALS['TYPO3_REQUEST']->getAttribute('frontend.controller') or with $GLOBALS['TSFE']. Both is valid in my eyes - but both should be avoided if possible.

Reply:
Okay, it is available since TYPO3 v9. Interesting, didn't know about that before, stumpled upon it recently.